src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -466,9 +466,9 @@ typename MatType::elem_type FFN<
 {
   typename MatType::elem_type res = 0;
   res += EvaluateWithGradient(parameters, 0, gradient, 1);
+  MatType tmpGradient(gradient.n_rows, gradient.n_cols);
```

Comment:
Suggest to change this constructor to explicitly specify the fill type:

    MatType tmpGradient(gradient.n_rows, gradient.n_cols, arma::fill::none);

The `arma::fill::none` specifier means don't zero-out the elements at construction, which saves a bit of time.  If this is what's actually intended, it's better to be explicit about it.

Otherwise, change the fill type to `arma::fill::zeros`, so there are no surprises later.  Without this specifier, the tmpGradient matrix can be full of garbage values.

(Prior to Armadillo 10.5, fill::none was the default initialisation operation.  Since Armadillo 10.5, fill::zeros is the default initialisation operation.  mlpack currently has the minimum required Armadillo version of 9.800).

Alternatively, perhaps `tmpGradient.zeros()` is required within the loop, to ensure that no garbage or leftover values are being used in each iteration.